src/LogNormal/LogNormalMath.sol

```diff
@@ -80,12 +81,10 @@ function computeYGivenL(
     uint256 S,
     LogNormalParams memory params
 ) pure returns (uint256 ry) {
-    int256 d2 = computeD2(S, params);
-    int256 cdf = Gaussian.cdf(d2);
-    uint256 unsignedCdf = toUint(cdf);
+    int256 d2 = computeD2({ S: S, params: params });
+    uint256 cdf = toUint(Gaussian.cdf(d2));
```

Comment:
This is incorrect, CDF can return a negative value. Maybe if d2 is constrained it will constrain the cdf output to be only positive, but if that's the case it should be commented and made clear

Reply:
In the prev iteration we did the same thing but in two steps, are you saying the previous iteration is wrong too?

Reply:
yes the cdf can return negative value so this will make all the negative return values 0

Reply:
i just dont remember if d2 is bounded in a way such that the cdf output is also bounded

Reply:
i don't think it is based on reading the code, if you look at computed2 it can return a negative value if halfSigmaSquared > ln(S/K) then it can be negative, I've just never encountered this scenario in the sims i guess

Reply:
no we have, remember it was a big bug. regardless, you are basically saying the cdf only outputs positive values, which is wrong

Reply:
talking about cdf function not the d2